datafusion/physical-expr/src/aggregate/count_distinct.rs

```diff
@@ -438,11 +446,292 @@ where
     }
 }
 
+#[derive(Debug)]
+struct StringDistinctCountAccumulator(Mutex<SSOStringHashSet>);
+impl StringDistinctCountAccumulator {
+    fn new() -> Self {
+        Self(Mutex::new(SSOStringHashSet::new()))
+    }
+}
+
+impl Accumulator for StringDistinctCountAccumulator {
+    fn state(&self) -> Result<Vec<ScalarValue>> {
+        // TODO this should not need a lock/clone (should make
+        // `Accumulator::state` take a mutable reference)
```

Comment:
In order to avoid the copy the accumulator needs to be &mut here, but it is not. This is worth its own ticket I think

Reply:
https://github.com/apache/arrow-datafusion/pull/8925